src/button-card.ts

```diff
@@ -0,0 +1,569 @@
+          border-bottom-left-radius: 2px;
+          border-bottom-right-radius: 2px;
+          border-top-left-radius: 2px;
+          border-top-right-radius: 2px;
+        }
+        @keyframes blink{
+          0%{opacity:0;}
+          50%{opacity:1;}
+          100%{opacity:0;}
+        }
+        @-webkit-keyframes rotating /* Safari and Chrome */ {
+          from {
+            -webkit-transform: rotate(0deg);
+            -o-transform: rotate(0deg);
+            transform: rotate(0deg);
+          }
+          to {
+            -webkit-transform: rotate(360deg);
+            -o-transform: rotate(360deg);
+            transform: rotate(360deg);
+          }
+        }
+        @keyframes rotating {
+          from {
+            -ms-transform: rotate(0deg);
+            -moz-transform: rotate(0deg);
+            -webkit-transform: rotate(0deg);
+            -o-transform: rotate(0deg);
+            transform: rotate(0deg);
+          }
+          to {
+            -ms-transform: rotate(360deg);
+            -moz-transform: rotate(360deg);
+            -webkit-transform: rotate(360deg);
+            -o-transform: rotate(360deg);
+            transform: rotate(360deg);
+          }
+        }
+        .rotating {
+          -webkit-animation: rotating 2s linear infinite;
+          -moz-animation: rotating 2s linear infinite;
+          -ms-animation: rotating 2s linear infinite;
+          -o-animation: rotating 2s linear infinite;
+          animation: rotating 2s linear infinite;
+        }
+      `;
+  }
+
+  protected render(): TemplateResult | void {
+    let state: HassEntity | null = null;
+    if (this.config!.entity) {
+      state = this.hass!.states[this.config!.entity];
+    }
+    const configState = this.testConfigState(state)
+    if (this.config)
+      switch (this.config.color_type) {
+        case 'blank-card':
+          return this.blankCardColoredHtml(state);
+        case 'label-card':
+        case 'card':
+          return this.cardColoredHtml(state, configState);
+        case 'icon':
+        default:
+          return this.iconColoredHtml(state, configState);
+      }
+  }
+
+
+  private getFontColorBasedOnBackgroundColor(backgroundColor: string): string {
+    let localColor = backgroundColor;
+    if (backgroundColor.substring(0, 3) === "var") {
+      localColor = window.getComputedStyle(document.documentElement)
+        .getPropertyValue(backgroundColor.substring(4).slice(0, -1)).trim();
+    }
+    const colorObj = new TinyColor(localColor);
+    let fontColor = ''; // don't override by default
+    if (colorObj.isValid && colorObj.getLuminance() > 0.5) {
+      fontColor = 'rgb(62, 62, 62)'; // bright colors - black font
+    } else {
+      fontColor = 'rgb(234, 234, 234)';// dark colors - white font
+    }
+    return fontColor;
+  }
+
+  private testConfigState(state: HassEntity | null): undefined | StateConfig {
+    if (!state || !this.config || !this.config.state) {
```

Comment:
```suggestion
    if (!state || !this.config.state) {
```
We're already testing this in `render`